primitives/io/src/lib.rs

```diff
@@ -1290,6 +1290,17 @@ pub trait Allocator {
 	}
 }
 
+/// WASM-only interface which allows for aborting the execution in case
+/// of an unrecoverable error.
+#[runtime_interface(wasm_only)]
+pub trait PanicHandler {
+	/// Aborts the current execution with the given error message.
+	#[trap_on_return]
+	fn abort_on_panic(&mut self, message: &str) {
```

Comment:
Why not call it `abort`? The host interface isn't concerned with the fact that a panic happened, in fact it shouldn't even concerned with the fact that the runtime is written in Rust.

Reply:
Hm, well, we can still change it. I don't really have a strong opinion *either* way and don't really want to get into bikeshedding the name, so I'm fine with whatever's the consensus.